app/helpers/measure_unit_helper.rb

```diff
@@ -0,0 +1,12 @@
+module MeasureUnitHelper
+  def presented_unit
+    @presented_unit ||= {
+      answer: user_session.measure_amount[component.unit.downcase.to_s],
```

Comment:
This is encapsulating logic about how to get a key suitable for `measure_amount` from a `component.unit` - that logic could be moved to the data object, eg `user_session.measure_amount[component.unit_key]`

Reply:
Yeah, makes sense. I've moved it to a concern :)